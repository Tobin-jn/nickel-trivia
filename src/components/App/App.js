import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';
import './App.css';
import Categories from '../../containers/Categories';
import DailyQuestion from '../../containers/DailyQuestion';
import TriviaGame from '../../containers/TriviaGame';

class App extends Component {
  render() {
    return (
      <div className="App">
        <h1 className="title">Nickle Trivia</h1>
        <button className="daily-trivia-btn">Trivia of the Day</button>
        <button className="trivia-game">Trivia Game</button>
        <Categories />
        <DailyQuestion />
        <TriviaGame />

      </div>
    );
  }
}

export default App;

        // <NavLink className='trivia-of-day-btn' to='/dailycategory'>
        //   Daily Trivia Question
        // </NavLink>
        // <NavLink className='question-generator-btn' to='/categories'>
        //   Trivia Question Generator
          
        // </NavLink>

// <NavLink to='/about'>About</NavLink>
