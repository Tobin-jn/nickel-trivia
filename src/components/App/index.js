import React from 'react';
import { Redirect, Route, Switch } from 'react-router-dom';
import Home from '../Home'
import Categories from '../../containers/Categories';
import TriviaGame from '../../containers/TriviaGame';
import DailyQuestion from '../../containers/DailyQuestion';
import './App.css';

const App = props => {
  return (
    <div className="App">
      <Switch>
        <Route exact path='/' component={ Home } />
        <Route exact path='/dailyquestion' component={ DailyQuestion } />
        <Route exact path='/triviagame' component={ TriviaGame } />
        <Redirect to='/' />
      </Switch>
    </div>
  );
}

export default App;