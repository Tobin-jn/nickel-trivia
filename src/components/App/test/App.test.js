/* eslint-disable */

import React from "react";
import { shallow } from "enzyme";
import App from '../App';

describe("App", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<App />);
  });

  it("should exist", () => {
    expect(wrapper).toBeDefined();
  });

  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});

// describe('mapStateToProps', () => {
//   it("should return a props object", () => {

//   })
// })

// describe('mapDispatchToProps', () => {
//   it('should call dispatch with...', () => {

//   })
// })
