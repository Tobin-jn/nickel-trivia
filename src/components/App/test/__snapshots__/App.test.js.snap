// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="title"
        >
          Nickle Trivia
        </h1>,
        <button
          className="daily-trivia-btn"
        >
          Trivia of the Day
        </button>,
        <button
          className="trivia-game"
        >
          Trivia Game
        </button>,
        <Connect(Categories) />,
        <Connect(DailyQuestion) />,
        <Connect(TriviaGame) />,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Nickle Trivia",
          "className": "title",
        },
        "ref": null,
        "rendered": "Nickle Trivia",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Trivia of the Day",
          "className": "daily-trivia-btn",
        },
        "ref": null,
        "rendered": "Trivia of the Day",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Trivia Game",
          "className": "trivia-game",
        },
        "ref": null,
        "rendered": "Trivia Game",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="title"
          >
            Nickle Trivia
          </h1>,
          <button
            className="daily-trivia-btn"
          >
            Trivia of the Day
          </button>,
          <button
            className="trivia-game"
          >
            Trivia Game
          </button>,
          <Connect(Categories) />,
          <Connect(DailyQuestion) />,
          <Connect(TriviaGame) />,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Nickle Trivia",
            "className": "title",
          },
          "ref": null,
          "rendered": "Nickle Trivia",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Trivia of the Day",
            "className": "daily-trivia-btn",
          },
          "ref": null,
          "rendered": "Trivia of the Day",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Trivia Game",
            "className": "trivia-game",
          },
          "ref": null,
          "rendered": "Trivia Game",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
